core/shared/src/main/scala/zio/stm/TArray.scala

```diff
@@ -330,6 +330,12 @@ final class TArray[A] private[stm] (private[stm] val array: Array[TRef[A]]) exte
       }
     }
 
+  /**
+   * Returns the size of the array.
+   */
+  def size: Int =
```

Comment:
Should we lift this value to STM? It would be consistent with the API, although this version has better performance as it removes one mapping.

Reply:
I thought about it or adding a note. It seems strange because the size is a constant so it doesn't require an effect to evaluate but don't feel strongly.

Reply:
Let's keep it this way then.